File: BackEnd/Models/Clan.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace Models
{
    [Table("Clan")]
    public class Clan
    {
        [Key]
        public int ID { get; set; }    

        [Required]
        [MaxLength(20)]
        public string  Ime { get; set; }

        [Required]
        [MaxLength(20)]
        public string Prezime { get; set; }

        //[RegularExpression("[A-Z0-9._%+-]+@[A-Z0-9.-]+.[A-Z]{2,}")]  // email regex 
        [Required]
        [MaxLength(30)]
        public string Email { get; set; }

        [Required]
        public Trener trener {get;set;}   // Clan ima jednog trenera 

        [Required]
        public Clanarina clanarina {get;set;} // Clan ima jednu clanarinu
    
        [JsonIgnore]
        public List<Termin> termin {get;set;} 

        [Required]
        public Teretana teretana{get;set;}

    }
}

File: FrontEnd/Clan.js
export class Clan{


    constructor(id,ime,prezime,email,trener,clanarina,teretana) {
        this.id=id;
        this.ime=ime;
        this.prezime=prezime;
        this.email=email;
        this.trener=trener;
        this.clanarina=clanarina;
        this.teretana = teretana;
    }

    crtaj(host){

        var tr = document.createElement("tr");
        host.appendChild(tr);

        /*var el = document.createElement("td");
        el.innerHTML=this.id;
        tr.appendChild(el);*/

        /*let el = document.createElement("td");
        el.innerHTML=this.brKartice;
        tr.appendChild(el);*/

        let el = document.createElement("td");
        el.innerHTML=this.ime;
        tr.appendChild(el);

        el = document.createElement("td");
        el.innerHTML=this.prezime;
        tr.appendChild(el);

        el = document.createElement("td");
        el.innerHTML=this.email;
        tr.appendChild(el);

        el = document.createElement("td");
        el.innerHTML=this.trener;
        tr.appendChild(el);

        el = document.createElement("td");
        el.innerHTML=this.clanarina;
        tr.appendChild(el);

        el = document.createElement("td");
        el.innerHTML=this.teretana;
        tr.appendChild(el);

    }
}